docs/INSTALL.rst

```diff
@@ -286,7 +286,16 @@ result in a "all tests passed" mesage::
  PASSED (skips=7, expectedFailures=3, successes=1176)
  __________________________ summary ___________________________________
    py27: commands succeeded
-   congratulations :) 
+   congratulations :)
+
+You may also install VCS/git hooks to run linters and code checks to catch
+common errors before each commit and to run the full self-test suite to find
+less obvious regressions before each push to a remote.  Not that running the
+full self-test suite takes several minutes to expecting pushing to take some
+time.  If you can't or don't want to wait for the hooks in some cases, use
+the ``--no-verify`` option to ``git commit ...`` or ``$ git push ...``::
+
+  $ make install-vcs-hooks
```

Comment:
This stuff is probably better in the developer-guide; ideally INSTALL should be targeted at users. I do see some "hacking" stuff in there already, but if we merge this there are two places that talk about hooks. One should probably just be a pointer to the other.

Reply:
Also, probably no `$` in the inline example for consistency.

Reply:
As with #798 it should be explained how to install the hooks so that you only run them on push (not commit). Many developers do commits that aren't "ready" yet (or commonly develop without internet connections) so these developers probably don't want to run anything at all on `git commit`.

Reply:
> This stuff is probably better in the developer-guide

Does [this other comment on this PR](https://github.com/tahoe-lafs/tahoe-lafs/pull/830#issuecomment-699659701) address your concern here?  If not, shouldn't drafting a new document containing developer guidelines be a separate PR?

Reply:
Ah, I see that #798 has been merged since this change.  I'll move the documentation now that there are developer guidelines.

Reply:
> Also, probably no `$` in the inline example for consistency.

Oops, I missed the $ for one.  I'll add it, thanks.

I prefer to include as many context clues as possible in code snippets (inline or block) since it's often non-obvious and costs so little space to be much clearer.  For example `$ this-is-a-command-to-run-in-a-shell` while `./this-is-a-directory-path/` while `./this-is-a-file-path`.

Reply:
Okay, I don't have strong feelings either way (and don't know if we have a convention) but yeah they should be the same at least :) thanks